python/cudf/cudf/tests/test_nocuinit.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+
+location = Path(__file__)
+cpp_build_dir = location / ".." / ".." / ".." / ".." / ".." / "cpp" / "build"
+libintercept = (cpp_build_dir / "libcudfcuinit_intercept.so").resolve()
```

Comment:
What's the right way to reference this? Right now I'm assuming the build directory exists (because I didn't manage to wrangle cmake to install the library). Equally, however, I'm not sure we really want to install this library?

Reply:
Oh boy, this is fun. I don't think there is a perfect solution here. FWIW my approach to this in #11875 was to move building the preload lib out of the main libcudf build, build it separately as part of CI, and then just launch tests with the preload library directly from the CLI in CI. That functionality was disabled as part of the Jenkins->GHA migration. Given that you're working on this, it may be time to investigate how to reenable that functionality within GHA.

@robertmaynard do you think that preload libraries like this or the stream verification lib should be built within the main CMakeLists.txt for the library, or shipped along with the conda packages? I had avoided that mostly because in the end we need the paths to the library anyway in order to preload, so it's not a great fit, but I know others had expressed different opinions. Depending on what direction we take with that we will need to adapt the solution in this pytest for how the library is discovered I think.

Reply:
Presume that the stream verification lib is also a single library. My first thought had been to just compile to the `.so` as part of the test, referencing the source directory. But then I realised that I need someone to provide information about the compiler configuration and so forth.